Ira/my-app-video/src/components/home/Home.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {Header} from "../layout/header";
+import SearchFilmForm from "../shared/searchFilmForm";
+import {IFilm} from "../../interfaces/IFilm";
+import {Footer} from "../layout/footer";
+import {FilmList} from "../filmList";
+import {SearchResultSection} from "../searchResultSection";
+import {RouteComponentProps} from "react-router-dom";
+import './styles.scss';
+
+const queryString = require('query-string');
+
+interface IOwnProps {
+  stateSortByBtn: string,
+  result: number,
+  films: IFilm[],
+
+  showFullFilmInfo(id: number): () => void,
+
+  sortBy(e: string): void,
+}
+
+interface IPathname {
+  sortBy?: string,
+  searchBy?: string,
+}
+
+interface IHomeState {
+  searchValue: string,
+  searchBy: string,
+  pathParams: IPathname,
+}
+
+type HomeProps = RouteComponentProps & IOwnProps;
+
+export class Home extends React.Component<HomeProps, IHomeState> {
+  state: IHomeState = {
+    searchValue: '',
+    searchBy: '',
+    pathParams: {},
+  }
+
+  componentDidUpdate(prevProps: Readonly<HomeProps>, prevState: Readonly<IHomeState>, snapshot?: any) {
+
+    if (prevProps.stateSortByBtn !== this.props.stateSortByBtn) {
+      this.setState({
+        pathParams: {
+          sortBy: this.props.stateSortByBtn
+        }
+      });
+    }
+
+    if (prevState.searchBy !== this.state.searchBy) {
+      this.setState({
+        pathParams: {
+          searchBy: this.state.searchBy,
+        }
+      });
+    }
+  };
+
+  getFormState = ({searchValue, searchBy}: { searchValue: string, searchBy: string }): void => {
+    this.setState({
+      searchValue,
+      searchBy,
+    })
```

Comment:
WHY ??? All you have to do is to change URL, you don't need to keep these values in state.